### PR TITLE
Implement ship placement

### PR DESCRIPTION
**Pull request by Aqcurate**
Ship are now unique. You can only place 3 ships: a battleship, a destroyer, and a minesweeper.

Individual squares can be checked for equality.

Placement now checks for collision and boundary conditions.

See Issues #3 and #4.

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -1,23 +1,77 @@
...
 
+	@JsonProperty private List<Ship> ships;
+	@JsonProperty private List<Result> attacks;
+
 	/*
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
 	public Board() {
-		// TODO Implement
+		ships = new ArrayList<>();
```
Comment (KSARK) [inline src/main/java/cs361/battleships/models/Board.java:17]: Seems like the constructor will be called several times when trying to place any ship, which causes the `ships` array never contain any ship.

Comment (Aqcurate) [reply src/main/java/cs361/battleships/models/Board.java:17]: Huh, I'll have to test it later. In theory, when deseralizing the JSON string, the constructor is first called (resulting in an empty `ships` ArrayList). But then, Jackson should update the `ships` ArrayList with the correct information via reflection.

Can you show me what errors you are getting on your end?

Comment (KSARK) [reply src/main/java/cs361/battleships/models/Board.java:17]: On my end, the `ships` ArrayList does not get updated.
image

Comment (Aqcurate) [reply src/main/java/cs361/battleships/models/Board.java:17]: That's so strange. It works on my end.

debug

Can you try something for me? This is how I have testing configured currently.

In games.js, I commented out:
```
        //markHits(game.opponentsBoard, "opponent", "You won the game");
        //markHits(game.playersBoard, "player", "You lost the game");
```

In Game.java, I replaced the opponent ship placing line with:
```
opponentPlacedSuccessfully = opponentsBoard.placeShip(new Ship(ship.getShipName()), (new Random()).nextInt(10) + 1, 'A', false);
```

The placement works on my end with these edits.

Comment (KSARK) [reply src/main/java/cs361/battleships/models/Board.java:17]: Now it works, thank you.

File: src/main/java/cs361/battleships/models/Board.java [hunk 2]
```diff
@@ -38,13 +38,13 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 		}
 
 		// User cannot place outside of grid
-		if (x < 1 || x > 10 || y < 'A' || y > 'J') {
+		if (x < 1 || x > 10 || y < 'A' || y > 'K') {
```
Comment (KSARK) [inline src/main/java/cs361/battleships/models/Board.java]: The range of `x` should include 10. Same issue in line 46.

**Review by KSARK — CHANGES_REQUESTED** [review]

File: src/main/java/cs361/battleships/models/Board.java (hunk 2 above)
Comment (Aqcurate) [reply src/main/java/cs361/battleships/models/Board.java]: Fixed.

**Review by KSARK — APPROVED** [review]

**Comment by staslyakhov** [comment]
Looks good to me, except some of the files don't end with a newline. Do we want to fix that potentially?

**Review by staslyakhov — APPROVED** [review]

**Comment by Csedge** [comment]
Ok your code work correctly, that great, it can help us complete the project step by step.That's good

**Pull request merged**